tests/FakeItEasy.IntegrationTests/FakingClassesTests.cs

```diff
@@ -12,10 +11,13 @@ public class FakingClassesTests
         [Test]
         public void Should_be_able_to_get_a_fake_value_of_uri_type()
         {
-            using (Fake.CreateScope(new NullFakeObjectContainer()))
-            {
-                A.Fake<Uri>();
-            }
+            // Arrange
+
+            // Act
+            var fake = A.Fake<Uri>();
+
+            // Assert
+            Fake.GetFakeManager(fake).Should().NotBeNull("because we should be able to create a fake Uri");

```

Comment:
> Is there a possible code path for A.Fake<T>() to return null?

I don't think so. It would throw an exception rather than returning null.

> the assertion should revert to asserting that no exception is thrown

The problem is that you can't (usually) catch a `StackOverflowException`... so it would be a bit pointless.